Class-Code/OOP/blehrer/sketch.js

```diff
@@ -13,46 +13,68 @@
 */
 
 class Bubble {
-    constructor(_x, _y, _r) {
+    constructor(_x, _y, _r, _speed, _color) {
         this.x = _x;
         this.y = _y;
         this.r = _r;
+        this.angle = random(0, 2 * PI);
+        this.speed = _speed;
+        this.color = _color;
     }
     show() {
-        fill(255);
+        push();
+        colorMode(HSB);
+        fill(this.color, 255, 255);
         ellipse(this.x, this.y, this.r)
+        pop();
     }
     move() {
-        this.x += random(-width / 100, width / 100);
-        this.y += random(-height / 100, height / 100);
+        this.x += this.speed * cos(this.angle);
+        this.y += this.speed * sin(this.angle);
+        this.angle += random(-PI / 6, PI / 6);
     }
-    pop() {
-        if(this.x >= width || this.x <= 0) {
-            this.r = 0;
+    edge() {
+        if (this.x >= width || this.x <= 0) {
+            this.angle *= -1;
             return;
         }
-        if(this.y >= height || this.y <= 0) {
-            this.r = 0;
+        if (this.y >= height || this.y <= 0) {
+            this.angle *= -1;
+        }
+    }
+    touch(index) {
+        for (let i = 0; i < bubbles.length; ++i) {
```

Comment:
I'm sure that there are people who would say that this method is great and some that might say the opposite. Personally, I think it's not good practice to call a global variable in your method. One of the key ideas in OOP is Encapsulation or the idea that the object is sort of self-sufficient. It should know how to check if it's touching another object, but to be able to check itself against a  global array might be a bit much. Also, if you ever change the name of the array you're going to have to go in and hard code that back in the object